src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -734,6 +734,28 @@ private Task BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
 
         public abstract int Read([In, Out] byte[] buffer, int offset, int count);
 
+        public virtual int Read(Span<byte> destination)
+        {
+            if (destination.Length == 0)
+            {
+                return 0;
+            }
+
+            ArrayPool<byte> pool = ArrayPool<byte>.Shared;
+            byte[] buffer = pool.Rent(destination.Length);
```

Comment:
Is there a significant advantage in getting the pool as a local rather than just calling ArrayPool<byte>.Shared.Rent()?